### PR TITLE
Add controls list tab to compliance reporting view

### PR DESCRIPTION
**Pull request by scottopherson**
### :nut_and_bolt: Description: What code changed, and why?

Add controls list tab to the compliance reporting view.

c

### :chains: Related Resources

https://github.com/chef/automate/issues/1310

### :+1: Definition of Done

A controls tab is available at `https://a2-dev.test/compliance/reports/controls` that displays a filterable list of controls.

### :athletic_shoe: How to Build and Test the Change

1. Navigate to https://a2-dev.test/compliance/reports/controls.
2. See list of controls.
3. Modify report filters via searchbar and/or other filter controls in reporting view.
4. See list of controls update accordingly.

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### CODE REVIEW

**Comment by jonong1972** [comment]
@scottopherson Is it possible to add a Profile Column to the list?  In the mock up I originally had the Profile and Profile version listed below the Control Name.  But I totally forgot about the Control ID and Control Description as 2 separate items.

For now, Can we just add a column between Control Name and Impact that is labeled "Profile"?

And then show the:
<Profile name>, <version number>

Example:
DevSec SSH Baseline, 2.2.0

**Comment by vjeffrey** [comment]
one more thing:
could you please apply the _profile_ AND _control_ filter when the user wants to filter by control?  this will enable deep filtering:
```
# apply a filter when the "filter by" button is clicked 
curl --silent --insecure -H "api-token: $A2_TOKEN" https://a2-dev.test/api/v0/compliance/reporting/controls -d '{"filters":[{"type":"profile_id","values":["1de944869a847da87d3774feaacb41829935a2f46b558f7fc34b4da21586ae27"]},{"type":"control","values":["/etc/group must exist"]},{"type":"start_time","values":["2019-09-15T00:00:00Z"]},{"type":"end_time","values":["2019-09-25T23:59:59Z"]}]}'
```

**Comment by scottopherson** [comment]
d

File: components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-controls/reporting-controls.component.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      .filter(v => v !== control.id).concat(control.id);
+
+    this.reportQuery.setFilterTitle(controlTypeName, control.id, control.title);
+
+    this.router.navigate([], {queryParams});
+  }
+
+  removeFilter(control) {
+    const filter = this.filterFor(control);
+    const typeName = filter.type.name;
```
Comment (vjeffrey) [inline components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-controls/reporting-controls.component.ts]: also remove profile filter? looks like only the control filter is being removed right now

Comment (scottopherson) [reply components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-controls/reporting-controls.component.ts]: @vj I wasn't sure. What if the user had already set a profile filter previously? Via searchbar or profiles tab, not via clicking on a control.

Comment (vjeffrey) [reply components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-controls/reporting-controls.component.ts]: ooooh, that's a good point! yes - that makes sense - ignore me!

i was worried that the user would click on a control and up with double profile filters or a control filter that doesn't match the profile, but that's not actually possible b/c the control list would only reflect controls in that profile.

thank you!!!

**Comment by vjeffrey** [comment]
got the ok frrom @jonong1972 to merge this without the total count from @rickmarry

**Pull request merged**